apitest.js

```diff
@@ -43,25 +56,84 @@ function getStreaminginfo(id) {
         .then(response => response.json())
         .then((data) => {
             console.log(data)
-            renderdata(data)
+            /*Checks to make sure the Watchmode API has any streaming options available for the chosen movie in its database*/
+            /*For example, Watchmode has the movie Neo Ned in its database, has an ID for it, but has no streaming links for it available (no subscription, rental or buying options)*/
+            if (data.length != 0) {
+                rendersubdata(data)
+                renderrentdata(data)
+                renderbuydata(data)
+            } else { errormessage.textContent = "No Streaming Options Found! Please try searching for a different movie." }
         })
         .catch(err => {
             console.error(err);
         });
 }
 
-function renderdata(data) {
-    var webStreamingServices = data
-        .filter(movie => { 
-            if (movie.region == "US" && movie.type =="sub" && movie.web_url != undefined){
-                return true;
-            } }
-        )
-        webStreamingServices.forEach(myFunction);
-        function myFunction(value){
-    console.log(value.web_url)}
+function rendersubdata(data) {
+    var subscriptionoptions = data
+        .filter(movie => {
+            if (movie.region == "US" && movie.type == "sub" && movie.web_url != undefined) { return true; }
+        })
+    console.log(subscriptionoptions);
+    /*Checks to make sure the Watchmode API has any SUBSCRIPTION streaming options available for the chosen movie in its database*/
+    /*For example, Watchmode has the movie SpiceGirls in its database, has an ID for it, you can buy or rent movie, but no subscription streaming links available*/
+    if (subscriptionoptions.length != 0) {
+        subscriptionoptions.forEach(value => {
+            var list = document.createElement("ul");
+            var link = document.createElement("li");
+            link.textContent = value.web_url
+            list.appendChild(link);
+            suboptions.appendChild(list);
+        });
+        subheading.textContent = "Subscription Streaming Links Available:"
+    } else {
+        suberror.textContent = "No Subscription Services Links Available"
```

Comment:
It will be more usable if we make this function return true or false here or just return the array. In general it is a good practice to make function only do one thing. This function checks for the subscriptions and displays them. It is better to have two functions one checks and another prints.